app/assets/javascripts/visualizations/highvis/photos.coffee

```diff
@@ -52,39 +46,69 @@ $ ->
         $(canvas).html("")
 
         # load the Handlebars templates
+        grpTemp = HandlebarsTemplates['visualizations/photo/group']
         picTemp = HandlebarsTemplates['visualizations/photo/pic']
         lbTemp = HandlebarsTemplates['visualizations/photo/lightbox']
 
-        # add each photo to the canvas
+        # group the media
+        groupedMedia = {}
+        groupedMedia['all'] = []
+        for _, dset of data.metadata
+          label = "#{dset.name}(#{dset.dataset_id})"
+          groupedMedia[label] = []
+          for pic in dset.photos
+            groupedMedia[label].push pic
+            groupedMedia['all'].push pic
+
+        # determine which data sets are selected based on the groupSelection
+        # also, sort them because they move around otherwise
+        selectedGroups = data.groupSelection.map (x) ->
+          [x, data.groups[x]]
+        selectedGroups.sort (x, y) ->
+          x[0] > y[0]
+
+        # create the groups to put the photos in
         id = 0
-        for dsKey,dset of data.metadata
-          if dset.photos.length > 0
-            for picKey,pic of dset.photos
-              context =
-                p_id:   'pic-' + id
-                tn_src: pic.tn_src
-                src:    pic.src
-                p_name: pic.name
-                d_name: dset.name
-                d_id:   dset.dataset_id
-
-              $(canvas).append picTemp(context)
-
-              $('#pic-' + id).data('context', context)
-              $('#pic-' + id).click ->
-                ctx = $(@).data('context')
-                $(canvas).append lbTemp(ctx)
-
-                $('#target-img').modal
-                  keyboard: true
-                $('#target-img').on 'hidden.bs.modal', ->
-                  $('#target-img').remove()
-
-              id++
+        for group in selectedGroups
+          if groupedMedia[group[1]].length == 0
```

Comment:
What if groupedMedia[group[1]] is undefined? (this is occurring in my localhost)